pkg/sql/schemachanger/scbuild/scbuild_test: TestBuildDataDriven failed

pkg/sql/schemachanger/scbuild/scbuild_test.TestBuildDataDriven [failed](https://tanzanite.cluster.engflow.com/invocations/default/b55acd1a-eb0f-44fd-9c4e-07e32e8b2072?testReportRun=5&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NjaGVtYWNoYW5nZXIvc2NidWlsZDpzY2J1aWxkX3Rlc3Q=) on release-24.1 @ [e5f8e9cc5c6fd09ac54a8cc087f9855e3e6d772d](https://github.com/cockroachdb/cockroach/commits/e5f8e9cc5c6fd09ac54a8cc087f9855e3e6d772d):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestBuildDataDriven (14m57s)
	TestBuildDataDriven/drop_function (14m28s)
	TestBuildDataDriven/drop_function/sql_dependencies (14m28s)

```
Stack: 

```
goroutine 119179 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        - [[Owner:{DescID: 110}, PUBLIC], ABSENT]
          {descriptorId: 110, owner: root}
        - [[UserPrivileges:{DescID: 110, Name: admin}, PUBLIC], ABSENT]
          {descriptorId: 110, privileges: "2", userName: admin, withGrantOption: "2"}
        - [[UserPrivileges:{DescID: 110, Name: public}, PUBLIC], ABSENT]
          {descriptorId: 110, privileges: "1048576", userName: public}
        - [[UserPrivileges:{DescID: 110, Name: root}, PUBLIC], ABSENT]
          {descriptorId: 110, privileges: "2", userName: root, withGrantOption: "2"}
        - [[FunctionBody:{DescID: 110}, PUBLIC], ABSENT]
          {body: "SELECT a FROM t;\nSELECT b FROM t@t_idx_b;\nSELECT c FROM t@t_idx_c;\nSELECT a FROM v;\nSELECT b'@':::@100108;\nSELECT nextval(105:::REGCLASS);", functionId: 110, lang: {lang: SQL}, usesSequenceIds: [105], usesTables: [{columnIds: [1], tableId: 104}, {columnIds: [2], indexId: 2, tableId: 104}, {columnIds: [3], indexId: 3, tableId: 104}], usesTypeIds: [108, 109], usesViews: [{columnIds: [1], viewId: 107}]}
    datadriven.go:144: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/schemachanger/scbuild/scbuild_test_/scbuild_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/schemachanger/scbuild/testdata/create_function:44:
        build [0 args]
        CREATE FUNCTION f(a notmyworkday) RETURNS INT VOLATILE LANGUAGE PLpgSQL AS $$
          BEGIN
          SELECT a FROM t;
          SELECT b FROM t@t_idx_b;
          SELECT c FROM t@t_idx_c;
          SELECT a FROM v;
          SELECT 'Monday'::notmyworkday;
          RETURN nextval('sq1');
          END
        $$;
        ----
        - [[Function:{DescID: 111}, PUBLIC], ABSENT]
          {functionId: 111, params: [{class: {}, name: a, type: {closedTypeIds: [108, 109], type: {family: EnumFamily, oid: 100108, udtMetadata: {arrayTypeOid: 100109}}}}], returnType: {type: {family: IntFamily, oid: 20, width: 64}}}
        - [[SchemaChild:{DescID: 111, ReferencedDescID: 101}, PUBLIC], ABSENT]
          {childObjectId: 111, schemaId: 101}
        - [[FunctionName:{DescID: 111}, PUBLIC], ABSENT]
          {functionId: 111, name: f}
        - [[FunctionVolatility:{DescID: 111}, PUBLIC], ABSENT]
          {functionId: 111, volatility: {volatility: VOLATILE}}
        - [[Owner:{DescID: 111}, PUBLIC], ABSENT]
          {descriptorId: 111, owner: root}
        - [[UserPrivileges:{DescID: 111, Name: admin}, PUBLIC], ABSENT]
          {descriptorId: 111, privileges: "2", userName: admin, withGrantOption: "2"}
        - [[UserPrivileges:{DescID: 111, Name: public}, PUBLIC], ABSENT]
          {descriptorId: 111, privileges: "1048576", userName: public}
        - [[UserPrivileges:{DescID: 111, Name: root}, PUBLIC], ABSENT]
          {descriptorId: 111, privileges: "2", userName: root, withGrantOption: "2"}
        - [[FunctionBody:{DescID: 111}, PUBLIC], ABSENT]
          {body: "BEGIN\nSELECT a FROM t;\nSELECT b FROM t@t_idx_b;\nSELECT c FROM t@t_idx_c;\nSELECT a FROM v;\nSELECT b'@':::@100108;\nRETURN nextval(105:::REGCLASS);\nEND;\n", functionId: 111, lang: {lang: PLPGSQL}, usesSequenceIds: [105], usesTables: [{columnIds: [1], tableId: 104}, {columnIds: [2], indexId: 2, tableId: 104}, {columnIds: [3], indexId: 3, tableId: 104}], usesTypeIds: [108, 109], usesViews: [{columnIds: [1], viewId: 107}]}
=== RUN   TestBuildDataDriven/create_sequence
=== RUN   TestBuildDataDriven/drop_function/sql_dependencies
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    conditional_wrap.go:185: 
        pkg/sql/schemachanger/scbuild/scbuild_test_test/pkg/sql/schemachanger/scbuild/builder_test.go:120: (TestBuildDataDriven.func1.3.1)
        	NOTICE: .NodeID() called via implicit interface StorageLayerInterface;
        HINT: consider using .StorageLayer().NodeID() instead.
    builder_test.go:151: created relation with id 104
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBuildDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-44224

--- a/pkg/sql/schemachanger/scbuild/BUILD.bazel
+++ b/pkg/sql/schemachanger/scbuild/BUILD.bazel
@@ -72,6 +72,7 @@ go_test(
         "main_test.go",
     ],
     data = glob(["testdata/**"]),
+    exec_properties = {"test.Pool": "large"},
     deps = [
         ":scbuild",
         "//pkg/base",

